### PR TITLE
fix: Fixed Backend Unit Test Cases

### PR DESCRIPTION
**Pull request by midhun-aot**
Fixed All backend unit test cases 
image

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-101-backend-sites.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-101-frontend-site-registry.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/sites/src/app/entities/siteHistoryLog.entity.ts [hunk 1]
```diff
@@ -21,6 +21,6 @@ export class HistoryLog extends BaseAuditEntity {
   siteId: string;
 
   @Field(() => GraphQLJSON)
-  @Column('jsonb', { name: 'snapshot_data' })
-  content: SaveSiteDetailsDTO;
+  @Column('jsonb', { name: 'data' })
+  content:  Record<string, any>
```
Comment (bolny) [inline backend/sites/src/app/entities/siteHistoryLog.entity.ts]: What is the purpose for this change?

File: backend/sites/src/app/resolvers/site/site.resolver.ts [hunk 2]
```diff
@@ -242,8 +242,6 @@ export class SiteResolver {
    */
   @Roles({
     roles: [
-      CustomRoles.External,
-      CustomRoles.Internal,
```
Comment (bolny) [inline backend/sites/src/app/resolvers/site/site.resolver.ts]: This seems reasonable, is it so that an existing unit test will pass?

File: backend/sites/src/app/resolvers/site/site.resolver.ts [hunk 3]
```diff
@@ -289,7 +287,7 @@ export class SiteResolver {
     }
   }
 
-  @Roles({ roles: [CustomRoles.Internal], mode: RoleMatchingMode.ANY })
+  @Roles({ roles: [CustomRoles.SiteRegistrar], mode: RoleMatchingMode.ANY })
```
Comment (bolny) [inline backend/sites/src/app/resolvers/site/site.resolver.ts]: Same as above..

File: backend/sites/src/app/services/site/site.service.ts [hunk 4]
```diff
@@ -1719,6 +1719,21 @@ export class SiteService {
...
+        whoCreated: userInfo ? userInfo.givenName : '',
+        whenCreated: new Date(),
+        whenUpdated: new Date(),
+        whoUpdated: userInfo ? userInfo.givenName : '',
+        siteId: site.siteId,
+      };
+  
+      await transactionalEntityManager.save(HistoryLog, historyLog);
+
+
```
Comment (bolny) [inline backend/sites/src/app/services/site/site.service.ts]: Is this change being made in order to pass an existing unit test?

File: backend/sites/src/migrations/1728425598626-master-script.ts
Comment (bolny) [inline backend/sites/src/migrations/1728425598626-master-script.ts:1]: Are these three migrations necessary for passing unit tests?

**Review by bolny — COMMENTED** [review]
Just some questions about some of the changes made since I don't see updates to the related test files.

File: backend/sites/src/app/entities/siteHistoryLog.entity.ts (hunk 1 above)
Comment (midhun-aot) [reply backend/sites/src/app/entities/siteHistoryLog.entity.ts]: This was due to branch created from a previous commit, but latest dev is merged to this branch and no longer its not part of the pull request.

File: backend/sites/src/app/services/site/site.service.ts (hunk 4 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/site/site.service.ts]: same here , this was due to branch created from a previous commit, but latest dev is merged to this branch and no longer its not part of the pull request.

**Review by bolny — APPROVED** [review]

**Review by nikhila-aot — APPROVED** [review]
